cylc/flow/network/scan.py

```diff
@@ -196,11 +196,13 @@ def re_compile_filters(patterns_owner=None, patterns_name=None):
     return (cres['owner'], cres['name'])
 
 
-def get_scan_items_from_fs(owner_pattern=None, reg_pattern=None):
+def get_scan_items_from_fs(
+        owner_pattern=None, reg_pattern=None, compile_reg=False,
```

Comment:
It we compile it in `cylc-print`, then I think the function gets simpler, as it receives an `owner_pattern`: `Pattern`, and a `reg_pattern` typed also `Pattern`. Alternatively, if we really need it, we still have the option to check the type of the `reg_pattern` or simply let it be recompiled (`re.compile` is [fine re-compiling a `Pattern`](https://github.com/python/cpython/blob/2f224a077a83ac9de8a12bb7dcc516642b8176d8/Lib/re.py#L295) as long as you don't pass any flag, but that's not really nice I guess).

But I think the simplest approach is to compile it in `cylc-print` if that's the only place using a string.

Reply:
Ah, sorry, I've commented on your review comments above before reading this, but it's nice to see you came to the same conclusion as I did! Agreed, compiling it in ``cylc-print`` is simplest (& cleanest) & would only do so once rather than N times.